features/app/components/AbstractApp.js

```diff
@@ -22,6 +31,30 @@ export class AbstractApp extends Component {
     }
 
     /**
+     * Init lib-jitsi-meet and create local participant when component is going
+     * to be mounted.
+     *
+     * @inheritdoc
+     */
+    componentWillMount() {
```

Comment:
If constructor fails, instance won't be created at all, and componentWillUnmount will not be called too.
If error is thrown inside ```componentWillMount``` component won't be rendered, and, potentially, app may stuck. But the same actually applies for any JavaScript code and for any other method and in 99,9% cases can be spotted and fixed during development/testing. Error handling in React seems to be a hot topic actually - https://github.com/facebook/react/issues/2461. 
And if we're talking exactly about ```componentWillMount``` method, I found in React code some stuff that will call unmount under some circumstances - https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L350-L360 and https://github.com/facebook/react/blob/master/src/renderers/shared/stack/reconciler/ReactCompositeComponent.js#L458-L490 respectively. So it seems we can define a method named ```unstable_handleError``` in which we can try to handle the error. Or even if not handle, just it presence (even as if empty function), will make ```componentWillUnmount``` to be called.